docs/StardustDocs/topics/readSqlFromCustomDatabase.md

```diff
@@ -0,0 +1,167 @@
+
+3. **Start the HSQLDB Server**:
+
+Launch a terminal or command prompt and execute the following command:
+
+```bash
+java -classpath lib/hsqldb.jar org.hsqldb.server.Server --database.0 file:hsqldb/demodb --dbname.0 testdb
+```
+
+## Implementing Custom Database Type Support
+
+To enable HSQLDB integration, implement a custom `DbType` by overriding required methods.
+
+
+**Create the HSQLDB Type**
+
+```kotlin
+import org.jetbrains.kotlinx.dataframe.io.TableColumnMetadata
+import org.jetbrains.kotlinx.dataframe.io.TableMetadata
+import org.jetbrains.kotlinx.dataframe.io.db.DbType
+import org.jetbrains.kotlinx.dataframe.schema.ColumnSchema
+import java.sql.ResultSet
+import java.util.Locale
+import kotlin.reflect.KType
+
+/**
+ * Represents the HSQLDB database type.
+ *
+ * This class provides methods to convert data from a ResultSet to the appropriate type for HSQLDB,
+ * and to generate the corresponding column schema.
+ */
+public object HSQLDB : DbType("hsqldb") {
+    override val driverClassName: String
+        get() = "org.hsqldb.jdbcDriver"
+
+    override fun convertSqlTypeToColumnSchemaValue(tableColumnMetadata: TableColumnMetadata): ColumnSchema? {
+        return null
+    }
+
+    override fun isSystemTable(tableMetadata: TableMetadata): Boolean {
+        val locale = Locale.getDefault()
+        fun String?.containsWithLowercase(substr: String) = this?.lowercase(locale)?.contains(substr) == true
+        val schemaName = tableMetadata.schemaName
+        val name = tableMetadata.name
+        return schemaName.containsWithLowercase("information_schema") ||
+            schemaName.containsWithLowercase("system") ||
+            name.containsWithLowercase("system_")
+    }
+
+    override fun buildTableMetadata(tables: ResultSet): TableMetadata =
+        TableMetadata(
+            tables.getString("TABLE_NAME"),
+            tables.getString("TABLE_SCHEM"),
+            tables.getString("TABLE_CAT"),
+        )
+
+    override fun convertSqlTypeToKType(tableColumnMetadata: TableColumnMetadata): KType? {
+        return null
+    }
+}
+```
+
+**Defining Helper Functions**
+
+Define utility functions to manage database connections and tables.
+
+```kotlin
+import java.sql.Connection
+
+const val URL = "jdbc:hsqldb:hsql://localhost/testdb"
+const val USER_NAME = "SA"
+const val PASSWORD = ""
+
+
+fun removeTable(con: Connection): Int {
+    val stmt = con.createStatement()
+    return stmt.executeUpdate("""DROP TABLE orders""")
+}
+
+fun createAndPopulateTable(con: Connection) {
+    var stmt = con.createStatement()
+    stmt.executeUpdate(
+        """CREATE TABLE IF NOT EXISTS orders (
+                        id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+                        item VARCHAR(50) NOT NULL,
+                        price DOUBLE NOT NULL,
+                        order_date DATE
+                    );
+                """.trimIndent()
```

Comment:
linted a bit weirdly, but overall fine, though I'd mention in the text above this code something like "For example purposes, we create a small function that can populate the table with a schema and some sample data."